test/extensions/filters/http/proto_api_scrubber/filter_config_test.cc

```diff
@@ -0,0 +1,484 @@
+#include "source/common/matcher/matcher.h"
+#include "source/extensions/filters/http/proto_api_scrubber/filter_config.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/server/factory_context.h"
+#include "test/proto/apikeys.pb.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "xds/type/matcher/v3/http_inputs.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ProtoApiScrubber {
+namespace {
+using ::envoy::extensions::filters::http::proto_api_scrubber::v3::MethodRestrictions;
+using ::envoy::extensions::filters::http::proto_api_scrubber::v3::ProtoApiScrubberConfig;
+using ::envoy::extensions::filters::http::proto_api_scrubber::v3::RestrictionConfig;
+using Http::HttpMatchingData;
+using xds::type::matcher::v3::HttpAttributesCelMatchInput;
+using MatchTreeHttpMatchingDataSharedPtr = Matcher::MatchTreeSharedPtr<HttpMatchingData>;
+using testing::NiceMock;
+
+// Structured test-case for parameterized tests of method name validation.
+struct MethodNameValidationTestCase {
+  std::string method_name;
+  absl::StatusCode expected_status_code;
+  std::string expected_status_message;
+};
+
+// A class to test method names which are provided in the filter config.
+class MethodNameValidation : public testing::TestWithParam<MethodNameValidationTestCase> {};
+
+TEST_P(MethodNameValidation, ValidateSingleMethodConfig) {
+  std::string filter_conf_string = absl::StrFormat(
+      R"pb(
+    restrictions: {
+      method_restrictions: {
+        key: "%s"
+        value: { }
+      }
+    }
+  )pb",
+      GetParam().method_name);
+  ProtoApiScrubberConfig proto_config;
+  ASSERT_TRUE(Protobuf::TextFormat::ParseFromString(filter_conf_string, &proto_config));
+  NiceMock<Server::Configuration::MockFactoryContext> factory_context;
+  absl::StatusOr<std::shared_ptr<ProtoApiScrubberFilterConfig>> filter_config =
+      ProtoApiScrubberFilterConfig::create(proto_config, factory_context);
+  EXPECT_EQ(filter_config.status().code(), GetParam().expected_status_code);
+  EXPECT_EQ(filter_config.status().message(), GetParam().expected_status_message);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    MethodNameValidationTestSuite, MethodNameValidation,
+    testing::ValuesIn<MethodNameValidationTestCase>({
```

Comment:
Interesting choice of using parameterized test here. I'm assuming the same test structure will work once other features will be supported. If not, consider reverting back to simple tests (TEST_F) that are also lightweight.